config/templates/cluster.yaml

```diff
@@ -165,8 +165,12 @@ kmsKeyArn: "{{.KMSKeyARN}}"
 # experimental:
 #   nodeDrainer:
 #     enabled: true
+#   awsEnvironment:
+#     enabled: true
+#     environment:
+#       CFNSTACK: '{ "Ref" : "AWS::StackId" }'
```

Comment:
There's definitely a trade-off made here. I like to not prevent smart people doing smart/dangerous things :smiling_imp: 

`CFNSTACK: 'AWS::StackId'` crossed my mind as well, but [sometimes](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-eip.html#aws-resource-ec2-eip-ref) you need resource attributes that can only be obtained with [Fn::GetAtt](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-getatt.html), not by `Ref`.

Better syntax checking could be done with something like:
```
environment:
  CFNSTACK:
    function: Ref
    parameter: AWS::StackId
```

I feel like this would ultimately complicate matters and not help all that much.

As for the `advanced:` section, i think there's some merit.. but i'm not sure yet. Isn't all of this advanced? :)